app/controllers/unattended_controller.rb

```diff
@@ -55,22 +53,26 @@ def hostgroup_template
   # Generate an action for each template kind
   # i.e. /unattended/provision will render the provisioning template for the requesting host
   def host_template
+    return head(:not_found) unless params[:kind].present?
+
+    return render_default_global_template if ipxe_request? && !@host
+    return render_local_boot_template if ipxe_request? && !@host.build?
```

Comment:
Just thinking loud: `ipxe_request?` could be generic like `request_kind?(:iPXE)` for the future.